apps/api/src/app/workflows-v2/workflow.controller.e2e.ts

```diff
@@ -193,6 +193,116 @@ describe('Workflow Controller E2E API Testing', () => {
       expect(idsDeduplicated.size).to.be.equal(10);
     });
   });
+
+  describe('Promote Workflow Permutations', () => {
```

Comment:
Given the fact that worfklow creation and promotion in general is a production critical part, I think it would make sense to cover `upsert-workflow.usecase.ts` with usecase specific tests to check if all entities are being created and updated correctly, by explicitly checking repositories.

I will create another task for this and add this in separate PR.